ple/games/vgdl/core.py

```diff
@@ -0,0 +1,743 @@
+                ss1, l1 = ss[g1]
+                for s1 in ss1:
+                    if not pygame.Rect((0,0), self.screensize).contains(s1.rect):
+                        effect(s1, None, self, **kwargs)
+                continue
+
+            # iterate over the shorter one
+            ss1, l1 = ss[g1]
+            ss2, l2 = ss[g2]
+            if l1 < l2:
+                shortss, longss, switch = ss1, ss2, False
+            else:
+                shortss, longss, switch = ss2, ss1, True
+
+            # score argument is not passed along to the effect function
+            score = 0
+            if 'scoreChange' in kwargs:
+                kwargs = kwargs.copy()
+                score = kwargs['scoreChange']
+                del kwargs['scoreChange']
+
+            # do collision detection
+            for s1 in shortss:
+                for ci in s1.rect.collidelistall(longss):
+                    s2 = longss[ci]
+                    if s1 == s2:
+                        continue
+                    # deal with the collision effects
+                    if score:
+                        self.score += score
+                    if switch:
+                        # CHECKME: this is not a bullet-proof way, but seems to work
+                        if s2 not in self.kill_list:
+                            effect(s2, s1, self, **kwargs)
+                    else:
+                        # CHECKME: this is not a bullet-proof way, but seems to work
+                        if s1 not in self.kill_list:
+                            effect(s1, s2, self, **kwargs)
+
+
+    def startGame(self, headless, persist_movie):
+        self._initScreen(self.screensize,headless)
+        pygame.display.flip()
+        self.reset()
+        clock = pygame.time.Clock()
+
+        win = False
+        i = 0
+        while not self.ended:
+            clock.tick(self.frame_rate)
+            self.time += 1
+            self._clearAll()
+
+            # gather events
+            pygame.event.pump()
+            self.keystate = pygame.key.get_pressed()
+
+            # load/save handling
+            if self.load_save_enabled:
+                from pygame.locals import K_1, K_2
+                if self.keystate[K_2] and self._lastsaved is not None:
+                    self.setFullState(self._lastsaved)
+                    self._initScreen(self.screensize,headless)
+                    pygame.display.flip()
+                if self.keystate[K_1]:
+                    self._lastsaved = self.getFullState()
+
+            # termination criteria
+            for t in self.terminations:
+                self.ended, win = t.isDone(self)
+                if self.ended:
+                    break
+            # update sprites
+            for s in self:
+                s.update(self)
+            # handle collision effects
+            self._eventHandling()
+            self._drawAll()
+            pygame.display.update(VGDLSprite.dirtyrects)

```

Comment:
PLE handles all the calls to `pygame.display.update`. Is it possible to remove the call to this?